src/NexusMods.DataModel/Games/GameLocationsRegister.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System.Diagnostics;
+using NexusMods.Paths;
+
+namespace NexusMods.DataModel.Games;
+
+/// <summary>
+/// Installation specific register of game locations <see cref="GameFolderType"/> and their resolved paths.
+/// <remarks>
+/// This class provides information about the locations, such as resolved paths, nested locations and top level parents.
+/// </remarks>
+/// </summary>
+public class GameLocationsRegister
+{
+    private readonly Dictionary<GameFolderType, GameLocationDescriptor> _locations = new();
+
+    /// <summary>
+    /// Obtain the resolved path for a <see cref="GameFolderType"/>.
+    /// </summary>
+    /// <param name="id">The <see cref="GameFolderType"/> to lookup</param>
+    public AbsolutePath this[GameFolderType id] => _locations[id].ResolvedPath;
+
+    /// <summary>
+    /// Construct a new instance of <see cref="GameLocationsRegister"/>.
+    /// </summary>
+    /// <remarks>
+    /// Computes the nested and top level parents relationships.
+    /// </remarks>
+    /// <param name="newLocations">A Dictionary of <see cref="GameFolderType"/>, <see cref="AbsolutePath"/> to register</param>
+    public GameLocationsRegister(IReadOnlyDictionary<GameFolderType, AbsolutePath> newLocations)
+    {
+        foreach (var (newId, newPath) in newLocations)
+        {
+            var newLocation = new GameLocationDescriptor(newId, newPath);
+
+            var isDuplicate = !_locations.TryAdd(newId, newLocation);
+
+            Debug.Assert(isDuplicate == false,
+                $"Duplicate location found for {newId} at {_locations[newId].ResolvedPath}: {newPath}");
+
+            foreach (var existingId in _locations.Keys)
+            {
+                if (existingId == newId) continue;
+
+                if (this[existingId].GetFullPathLength() < newPath.GetFullPathLength())
```

Comment:
This code appears to check if the existing path is a parent to the new path; however this is not clear at first.
Can you leave a comment in the code here? (for future maintainers)

Reply:
Yeah, good idea.